com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs

```diff
@@ -393,6 +405,74 @@ private void OnDestroy()
             }
         }
 
+        private SnapshotDespawnCommand GetDespawnCommand()
+        {
+            SnapshotDespawnCommand command;
+            command.NetworkObjectId = NetworkObjectId;
+            command.TickWritten = default; // will be reset in Despawn
+            command.TargetClientIds = default;
+
+            return command;
+        }
+
+        private SnapshotSpawnCommand GetSpawnCommand()
+        {
+            SnapshotSpawnCommand command;
+            command.NetworkObjectId = NetworkObjectId;
+            command.OwnerClientId = OwnerClientId;
+            command.IsPlayerObject = IsPlayerObject;
+            command.IsSceneObject = (IsSceneObject == null) || IsSceneObject.Value;
+
+            ulong? parent = NetworkManager.SpawnManager.GetSpawnParentId(this);
+            if (parent != null)
+            {
+                command.ParentNetworkId = parent.Value;
+            }
+            else
+            {
+                // write own network id, when no parents. todo: optimize this.
+                command.ParentNetworkId = command.NetworkObjectId;
+            }
+
+            command.GlobalObjectIdHash = HostCheckForGlobalObjectIdHashOverride();
+            // todo: check if (IncludeTransformWhenSpawning == null || IncludeTransformWhenSpawning(clientId)) for any clientId
+            command.ObjectPosition = transform.position;
+            command.ObjectRotation = transform.rotation;
+            command.ObjectScale = transform.localScale;
+            command.TickWritten = default; // will be reset in Spawn
```

Comment:
Same as above, I thought we record tick it happened not tick it was written out

Reply:
we do record the tick it was written, just deeper in the stack:

```
SnapshotSystem.Spawn() at com.unity.netcode.gameobjects/Runtime/Core/SnapshotSystem.cs:line 849
NetworkObject.SnapshotSpawn() at com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs:line 451
NetworkObject.SpawnInternal() at com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs:line 493

```

Reply:
```
        internal void Spawn(SnapshotSpawnCommand command)
        {
            command.TickWritten = m_CurrentTick;
```